test/simple/test-tls-cipher-list.js

```diff
@@ -60,11 +45,194 @@ function doTest(checklist, env, useswitch) {
+
+// Test the --cipher-list command line switch
+doTest('ABC', ['--cipher-list=ABC']);
+
+// Test the --enable-legacy-cipher-list and NODE_LEGACY_CIPHER_LIST envar
+['v0.10.38'].forEach(function(arg) {
+  var checklist = tls.getLegacyCiphers(arg);
+  // command line switch
+  doTest(checklist, ['--enable-legacy-cipher-list=' + arg]);
+  // environment variable
+  doTest(checklist, [], {'NODE_LEGACY_CIPHER_LIST': arg});
+});
+
+// Test the precedence order for the various options
+doTestPrecedence();
+
+// Test that we throw properly
+// invalid value
+assert.throws(function() {tls.getLegacyCiphers('foo');}, Error);
+// no parameters
+assert.throws(function() {tls.getLegacyCiphers();}, TypeError);
+// not a string parameter
+assert.throws(function() {tls.getLegacyCiphers(1);}, TypeError);
+// too many parameters
+assert.throws(function() {tls.getLegacyCiphers('abc', 'extra');}, TypeError);
+// ah, just right
+assert.doesNotThrow(function() {tls.getLegacyCiphers('v0.10.38');});
+
+// Test to ensure default ciphers are not set when v0.10.38 legacy cipher
+// switch is used. This is a bit involved... we need to first set up the
+// TLS server, then spawn a second node instance using the v0.10.38 cipher,
+// then connect and check to make sure the options are correct. Since there
+// is no direct way of testing it, an alternate createCredentials shim is
+// created that intercepts the call to createCredentials and checks the output.
+// The following server code was adopted from test-tls-connect-simple.
+// This spins up a server to verify that the connection is still able to
+// function with the default ciphers not set on the client side.
+
+// note that the following function is written out to a string and
+// passed in as an argument to a child node instance.
+var script = (
+  function() {
+    var tls = require('tls');
+    var orig_createCredentials = require('crypto').createCredentials;
+    var used_monkey_patch = false;
+    require('crypto').createCredentials = function(options) {
+      used_monkey_patch = true;
+      // since node was started with the --enable-legacy-cipher-list
+      // switch equal to v0.10.38, the options.ciphers should be
+      // undefined. If it's not undefined, we have a problem and
+      // the test fails
+      if (options.ciphers !== undefined) {
+        console.error(options.ciphers);
+        process.exit(1);
+      }
+      return orig_createCredentials(options);
+    };
+    var socket = tls.connect({
+      port: 0,
+      rejectUnauthorized: false
+    }, function() {
+      socket.end();
+      if (!used_monkey_patch) {
+        console.error('monkey patched createCredentials not used.');
+        process.exit(1);
+      }
+    });
+  }
+).toString();
+
+var test_count = 0;
+
+function doDefaultCipherTest(additional_args, env, opts) {
+  var options = {};
+  if (env) options.env = env;
+  var out = '', err = '';
+  additional_args = additional_args || [];
+  var args = additional_args.concat([
```

Comment:
Another way to achieve the same thing would be to use `fork` and to pass an argument to the child process that allows to easily identify what code to run in the parent and in the child processes. Something similar to what is done in https://github.com/joyent/node/blob/v0.10/test/simple/test-domain-top-level-error-handler-throw.js. A search with `grep -rHn "'child'" test/simple` will show you many tests where this pattern is used.

That would require to have this test in a separate file, and I think this would make things simpler and cleaner overall.